src/background/refreshToken.test.ts

```diff
@@ -48,32 +55,60 @@ const getCachedAuthDataMock = jest.mocked(getCachedAuthData);
 const setCachedAuthDataMock = jest.mocked(setCachedAuthData);
 const readRawConfigurationsMock = jest.mocked(readRawConfigurations);
 
-describe("refresh PKCE token", () => {
-  beforeEach(() => {
-    getCachedAuthDataMock.mockReset();
-    setCachedAuthDataMock.mockReset();
-    readRawConfigurationsMock.mockReset();
-    appApiMock.reset();
-    appApiMock.resetHistory();
+describe("refresh token argument validation", () => {
+  it("throws if integration id's don't match", async () => {
+    const integrationConfig = sanitizedIntegrationConfigFactory();
+
+    await expect(
+      refreshPKCEToken(googleIntegration, integrationConfig)
+    ).rejects.toThrow("Integration id and config service id do not match");
+  });
+
+  it("throws if integration is AA OAuth2", async () => {
+    const integrationConfig = sanitizedIntegrationConfigFactory({
+      serviceId: CONTROL_ROOM_OAUTH_INTEGRATION_ID,
+    });
+
+    await expect(
+      refreshPKCEToken(aaIntegration, integrationConfig)
+    ).rejects.toThrow(
+      `Use _refreshPartnerToken to refresh the ${CONTROL_ROOM_OAUTH_INTEGRATION_ID} token`
+    );
   });
 
   it("throws if integration configuration is not pkce", async () => {
-    const integrationConfig = sanitizedIntegrationConfigFactory();
+    const integrationConfig = sanitizedIntegrationConfigFactory({
+      serviceId: greenhouseIntegration.id,
+    });
 
     await expect(
       refreshPKCEToken(greenhouseIntegration, integrationConfig)
     ).rejects.toThrow(
       `Expected OAuth2 PKCE integration, but got ${greenhouseIntegration.id}`
     );
   });
+});
+
+describe.each(
+  // Create an array of arrays, where the first element is included in the test name,
+  // because using the integration object in the test name was breaking the test.
+  [googleIntegration, microsoftIntegration].map((x) => [x.id, x])
+)("refresh PKCE token for %s", (_, integration) => {
```

Comment:
I believe this would also work:

```suggestion
describe.each(
  [googleIntegration, microsoftIntegration]
)("refresh PKCE token for $id", (integration) => {
```

Up to you if you want to try it.